### ISSUE TITLE
Mailpit not working out of the box with sendmail in Craft CMS 

### ISSUE THREAD

**Issue by mandrasch** [ddev/ddev#5739]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Expected Behavior

- Install fresh Craft CMS (https://dev.to/mandrasch/install-craft-cms-v4-with-one-command-via-ddev-32da)
- Send a test mail in Settings > Email > Test
- E-Mail should be sendable and receiveable in Mailpit

Default settings of Craft CMS:

![image](https://github.com/ddev/ddev/assets/777278/df2d75ea-e9e4-46af-98c0-e0772a88b576)

SMTP works fine with these settings:

![image](https://github.com/ddev/ddev/assets/777278/f6df178e-a7b5-4c6d-b5b1-5cde8aa12f83)

.env values added by DDEV:
```
MAILPIT_SMTP_HOSTNAME=127.0.0.1
MAILPIT_SMTP_PORT=1025
```

### Actual Behavior

Error while sending the test mail with default settings, reported by Fabrice Delaneau in Discord (https://discord.com/channels/664580571770388500/1199651637686439936/1199651637686439936)

### Steps To Reproduce

- Install fresh Craft CMS (https://dev.to/mandrasch/install-craft-cms-v4-with-one-command-via-ddev-32da)
- Send a test mail in Settings > Email > Test

### Anything else?

Mailpit docs, multiple ways of doing this - e.g. symlink: https://mailpit.axllent.org/docs/install/sendmail/

Info by rfay:

> I think we used to have that set up for mailhog. But we did it with php.ini instead

**Comment by stasadev** [comment]
The values ​​in the dropdown list come from
https://github.com/craftcms/cms/blob/17164e79c7b3e86534491ff7687129cc1db79386/src/mail/transportadapters/Sendmail.php#L134-L145

![img](https://github.com/ddev/ddev/assets/24270994/eebed8d1-dd95-4516-a086-f5168974c4d0)

```php
$command = Craft::$app->getProjectConfig()->get('email.transportSettings.command');

return array_unique(array_filter([
    !str_starts_with($command, '$') ? $command : null,
    self::DEFAULT_COMMAND,
    ini_get('sendmail_path'),
]));
```

The first item in the dropdown (`null` by default), which can be overwritten with `ini_get('sendmail_path)` from DDEV.

```
$command = Craft::$app->getProjectConfig()->get('email.transportSettings.command');
```

@mandrasch, any ideas how to change `email.transportSettings.command` somewhere in the project config?

I've never used CraftCMS and looking at the documentation didn't help.

**Comment by mandrasch** [comment]
Hi @stasadev, ah interesting! Never had a deep look into this.

The standard way in craft is

1. Access dashboard in a local environment (with `allowAdminChanges` enabled)
2. Edit a setting like transport setting in dashboard, hit save
3. This is automatically stored in config/project/... yaml files by craft
4. And can be deployed to a production enviroment where you run `craft project-config/apply` 

(`allowAdminChanges` is disabled on production sites)

I just quick checked and saw that this is the default after install - nothing selected (should be null):

<img width="499" alt="image" src="https://github.com/ddev/ddev/assets/777278/7cf2f40d-982f-421f-a219-b3ebf0d52a53">

If you hit save for the first time, the selected setting will be added to project config as `email.transportSettings.command`:

<img width="505" alt="image" src="https://github.com/ddev/ddev/assets/777278/8b0e0422-daa2-4144-87ad-b7d69fd0168e">

So on a deeper look maybe this has to be better supported by craft to give ` ini_get('sendmail_path')` a higher priority than  `self::DEFAULT_COMMAND` when there was nothing selected? Not so much a DDEV issue? 🤔

**Comment by mandrasch** [comment]
(just fiy: Another option in craft is overriding some config values via `config/app.php` - this is what many devs do when they create project templates, but we should not do this via DDEV https://craftcms.com/docs/4.x/config/app.html#mailer)

Just had another look:

I'll open an issue at Craft CMS to ask if they could check here for an ini setting, as far as I see this is missing here (https://github.com/craftcms/cms/blob/17164e79c7b3e86534491ff7687129cc1db79386/src/mail/transportadapters/Sendmail.php#L119) 

```
    public function defineTransport(): array|AbstractTransport
    {
        // Replace any spaces with `%20` according to https://symfony.com/doc/current/mailer.html#other-options
        $command = Html::encodeSpaces(App::parseEnv($this->command) ?: self::DEFAULT_COMMAND);

        return [
            'dsn' => 'sendmail://default?command=' . $command,
        ];
    }
 ```

**Comment by rfay** [comment]
@mandrasch would you mind trying the mailpit-recommended technique? I can be done manually (temporarily) or with a Dockerfile.

**Comment by rfay** [comment]
IMO this should work OOTB without any changes, as long as Craft is using PHP properly. I wonder why it's not:

https://github.com/ddev/ddev/blob/0225d6f483adc606e9da1ffaf650c06d56f2882d/containers/ddev-php-base/ddev-php-files/etc/php/8.1/fpm/php.ini#L77

**Comment by mandrasch** [comment]
> IMO this should work OOTB without any changes, as long as Craft is using PHP properly. I wonder why it's not:
> 
> https://github.com/ddev/ddev/blob/0225d6f483adc606e9da1ffaf650c06d56f2882d/containers/ddev-php-base/ddev-php-files/etc/php/8.1/fpm/php.ini#L77

@stasadev pointed this out perfectly - Craft is proposing the php.ini value set by DDEV to users in Settings Screen, BUT users need to actively select and save it first. 

It's not the default (fallback) command which it should be. 

I'll take this to the craft community. :) 

Also an importanr note for support requests:

If devs use the overwrite in config/app.php, the Test Email Button in Craft Dashboard won't work anymore:

> Any changes you make to the Mailer component from config/app.php will not be reflected when testing email settings from Settings → Email.

https://craftcms.com/docs/4.x/config/app.html#mailer

<hr>

I think DDEV is doing it completely right currently by setting php.ini and providing the .env values 👍

**Comment by mandrasch** [comment]
fyi - posted here: https://github.com/craftcms/cms/discussions/14211

**Comment by rfay** [comment]
@mandrasch should we keep this open, or is this just a Craft problem?

**Comment by mandrasch** [comment]
Let's close it, only CraftCMS related. 👍

**Comment by mandrasch** [comment]
> @mandrasch should we keep this open, or is this just a Craft problem?

Closed it, thanks for reminder! It's CraftCMS-related from my point of view, thanks for investigating together. 

Learned some new things about how Craft handles things as well 👍

**Comment by webkp** [comment]
It appears that the issue has been fixed by Craft in version 4.8.0:
https://github.com/craftcms/cms/pull/14433

I thought I'd reference this here fyi and in case anyone else is looking for it.
